### PR TITLE
Add support for Python 3.7

### LINKED ISSUE #226: Not compatible with Python 3.7?
Linux, `CPython 3.7.3`.
By just importing `from iota import Iota` I get:

```
Traceback (most recent call last):
  File "/home/path/main.py", line 1, in <module>
    from iota import Iota
  File "/home/path/venv/lib/python3.7/site-packages/iota/__init__.py", line 38, in <module>
    from .api import *
  File "/home/path/venv/lib/python3.7/site-packages/iota/api.py", line 12, in <module>
    from iota.commands import BaseCommand, CustomCommand, core, \
  File "/home/path/venv/lib/python3.7/site-packages/iota/commands/__init__.py", line 13, in <module>
    import filters as f
  File "/home/path/venv/lib/python3.7/site-packages/filters/__init__.py", line 27, in <module>
    from filters.extensions import FilterExtensionRegistry
  File "/home/path/venv/lib/python3.7/site-packages/filters/extensions.py", line 11, in <module>
    from class_registry import EntryPointClassRegistry
  File "/home/path/venv/lib/python3.7/site-packages/class_registry/__init__.py", line 5, in <module>
    from .registry import *
  File "/home/path/venv/lib/python3.7/site-packages/class_registry/registry.py", line 33, in <module>
    class BaseRegistry(with_metaclass(ABCMeta, Mapping)):
  File "/home/path/venv/lib/python3.7/site-packages/six.py", line 827, in __new__
    return meta(name, bases, d)
  File "/usr/lib/python3.7/abc.py", line 126, in __new__
    cls = super().__new__(mcls, name, bases, namespace, **kwargs)
TypeError: type() doesn't support MRO entry resolution; use types.new_class()

Process finished with exit code 1
```

### PR DESCRIPTION
**Pull request by pdecol**
closes #226

### CODE REVIEW

**Comment by todofixthis** [comment]
Hey @pdecol this is awesome, thank you!  The test failure is because Travis CI's `trusty` distro doesn't know 3.7 is out of development; if you remove this line from the `.travis.yml` file, it should work:

https://github.com/iotaledger/iota.lib.py/blob/master/.travis.yml#L1

**Comment by pdecol** [comment]
Thanks! All tests passed

File: setup.py [hunk 1]
```diff
@@ -56,9 +56,13 @@
...
         ],
     },
+    
+    # The filter module is no longer maintained and does not support Python 3.7
+    # phx-filters is a fork that supports 3.7 and 3.8 but not 2.7
 
     install_requires=[
-        'filters',
+        'filters; python_version < "3.0"',
+        'phx-filters; python_version >= "3.5"',
```
Comment (todofixthis) [inline setup.py]: 👍🏻 Good thinking!!  Should we make the first one `filters; python_version < "3.5"`?

**Comment by todofixthis** [comment]
Awesome!  I'll have a look through the changes this week!

File: setup.py (hunk 1 above)
Comment (pdecol) [reply setup.py]: Yes. Although we do not officially support these versions, it may be better to have no gaps

**Comment by pdecol** [comment]
Sure. It is good to have things consistent and up to date. I updated the pull request. Thanks for your review!

**Comment by todofixthis** [comment]
Looks good!  Thanks @pdecol!

**Pull request merged**